packages/easy-email-extensions/src/AttributePanel/components/blocks/AdvancedTable/Operation/tableMenuConfig.ts

```diff
@@ -0,0 +1,279 @@
+              deletedIndex[deletedIndex.length - 1] - deletedIndex[0] + 1,
+            );
+          }
+        }
+      });
+
+      _this.changeTableData?.(_this.tableData);
+    },
+  },
+
+  deleteColumn: {
+    text: 'Delete selected columns',
+    icon: `<?xml version="1.0" standalone="no"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="icon" width="20px" height="20.00px" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg"><path fill="#595959" d="M925.996 99.038c-25.47-25.6-60.121-39.822-96.323-39.822H194.198c-75.12 0.13-136.016 61.026-136.145 136.146v635.345c0 36.073 14.351 70.723 39.822 96.323 25.6 25.73 60.25 40.081 96.323 40.081h635.475c36.072 0 70.723-14.351 96.323-39.951 25.6-25.6 39.951-60.122 39.951-96.324V195.362c0-36.073-14.351-70.724-39.951-96.324z m-365.77 494.287L512 545.228l-48.226 48.097-32.194-31.935 48.355-48.226-48.226-48.097 32.194-32.194L512 480.97l48.097-48.097 32.194 32.194-48.097 48.097 48.226 48.226-32.194 31.935zM103.434 195.362c0-24.049 9.568-47.192 26.635-64.13 17.066-17.066 40.08-26.634 64.129-26.634h136.145v226.91H103.434V195.361z m0 181.656h226.91V649.31h-226.91V377.02z m90.764 544.84c-24.049 0-47.192-9.567-64.13-26.634-17.066-17.066-26.634-40.08-26.634-64.258V694.69h226.91v227.168H194.197z m726.238-90.763c0 24.048-9.438 47.192-26.505 64.259-17.066 17.066-40.21 26.634-64.258 26.505H693.527V694.69h226.91v136.404z m0-181.786H693.527V377.02h226.91v272.29zM693.527 331.507V104.598h136.146c24.048 0 47.192 9.438 64.258 26.505 17.067 17.067 26.635 40.21 26.505 64.259v136.145H693.527z" /></svg>`,
+    handler() {
+      const _this = this as unknown as TableOperationMenu;
+      const { left, right } = _this.tableIndexBoundary;
+      _this.tableData.forEach(tr => {
+        const deleteIds = [] as number[];
+        for (let index = 0; index < tr.length; index++) {
+          const td = tr[index];
+          const tdLeft = tr[index].left || 0;
+          const tdRight = tr[index].right || 0;
+          const colSpan = td.colSpan || 1;
+          if (tdLeft > right) {
+            break;
+          }
+          if (tdLeft >= left && tdRight <= right) {
+            deleteIds.push(index);
+            break;
+          }
+          // operate one cell
+          if (tdLeft <= left && tdRight >= right) {
+            td.colSpan = colSpan - right - left - 1;
+            break;
+          }
+          // left insert cell
+          if (tdLeft > left && tdRight >= right) {
+            td.colSpan = colSpan - right - tdLeft - 1;
+            break;
+          }
+          // right insert cell
+          if (tdLeft < left && tdRight > left) {
+            td.colSpan = colSpan - tdRight - left - 1;
+            break;
+          }
+        }
+        if (deleteIds.length) {
+          tr.splice(deleteIds[0], deleteIds[deleteIds.length - 1] - deleteIds[0] + 1);
+        }
+      });
+      _this.changeTableData?.(_this.tableData);
```

Comment:
<img width="839" alt="image" src="https://github.com/user-attachments/assets/0b27f149-0dd4-40b4-b684-d7851e0d8685">
The operation here is a bit confusing. If the selected cell has colspan=2, then in the following row, the column should also delete colspan=2 or add colspan +1 to the cell on the right in the first row.

Reply:
fixed this bug